frontend/dispatcher.lua

```diff
@@ -1027,16 +1045,20 @@ end
 Calls the events in a settings list
 arguments are:
     1) the settings table
-    2) optionally a `gestures` object
+    2) execution management table: { qm_show = true|false} - forcibly show QM / run
+                                   { qm_anchor = ges.pos } - anchor position
+                                   { gesture = ges } - a `gestures` object
 --]]--
-function Dispatcher:execute(settings, gesture)
-    if settings.settings ~= nil and settings.settings.show_as_quickmenu == true then
-        return Dispatcher:_showAsMenu(settings, gesture)
+function Dispatcher:execute(settings, exec_props)
+    if ((exec_props == nil or exec_props.qm_show == nil) and settings.settings and settings.settings.show_as_quickmenu)
+            or (exec_props and exec_props.qm_show) then
```

Comment:
This conditional is a bit long and confusing to read, bur not sure of any way that this better